darshan-runtime/doc/darshan-runtime.txt

```diff
@@ -87,6 +87,9 @@ make install
   scheduler does not advertise the job ID) then you can specify `NONE` here.
   Darshan will fall back to using the pid of the rank 0 process if the
   specified environment variable is not set.
+* `--with-username-env=NAME`: this specifies the environment variable that Darshan
```

Comment:
Maybe we should have either a) a recipe (or recipe variation) for Slurm systems or b) some rationale for when/why you might want this?  This is a good fix, but I'm worried that people installing on systems with Slurm schedulers won't find it.

Reply:
This fix is primarily intended for Slurm systems which we also disable the use of `cuserid()`, which is just a subset of Cray/HPE system at this point. If `cuserid()` isn't disabled, this fix isn't likely to be useful.

I just scanned darshan-runtime docs and there really isn't any mention of the cuserid issue either. Maybe I can just update our existing Cray recipe to mention these issues and give an example there? That might be more helpful than burying in a note up in the long list of configure options.

We should also update the darshan-runtime package in Spack for next release to take advantage of this option when using the `+slurm` variant.

Reply:
Ah right.  I wonder if the problematic cuserid systems have about aged out at this point?

At any rate, that seems reasonable to put a mention of this in the existing Cray recipe.  Also agree with leveraging the +slurm spack variant.

Reply:
> Ah right. I wonder if the problematic cuserid systems have about aged out at this point?

I had hoped this was the case, but we actually ran into this problem again recently on NERSC Cori (see #699), so I feel like we should discourage usage of cuserid on Cray's indefinitely. It only affects statically-linked executables, which seem to increasingly be aged out themselves, but there will probably will always be apps forcing static linking for some reason -- we'll probably need to continue to be extra careful to avoid crashing apps in this context.

Reply:
Alright, I think this should be addressed via [f891648](https://github.com/darshan-hpc/darshan/pull/925/commits/f8916489cedc3f88eba3cbb84f38dc670f98488d)